src/layouts/Catalogue/Catalogue.js

```diff
@@ -1,185 +1,96 @@
-import React, {Component } from "react";
-import {Product, NavigationBar, Cart} from "../../components";
+import React, { useEffect, useState } from "react";
+import { Product, NavigationBar, Cart, AddProduct } from "../../components";
 import { ProductApi } from "../../api";
-import { Modal, Button, InputNumber, } from "antd";
-import 'antd/dist/antd.css';
+import "antd/dist/antd.css";
 import styled from "styled-components";
 
-const loading = (
-  <div className="suspense-page">
-    {" "}
-    Loading...{" "}
-  </div>
-);
-
-export default class Catalogue extends Component {
-  state = {
-    menuOpened: false,
-    products: [],
-    selectedProduct: {},
-    onLoading: false,
-    showModal: false,
-    showModalCart: false,
-    cart: [],
-    qty: 1,
-  };
-
-  getDataProducts() {
-    return ProductApi.get().then(res => res.json())
-    .then(res => {
-      this.setState({
-        products : res,
-      })
-    });
+export default function NewCatalogue() {
+  const [products, setProducts] = useState([]);
+  const [selectedProduct, setSelectedProduct] = useState({});
+  const [showModalProduct, setShowModalProduct] = useState(false);
+  const [showModalCart, setShowModalCart] = useState(false);
+  const [cart, setCart] = useState([]);
+  const [quantity, setQuantity] = useState(1);
+  const [itemTotal, setItemTotal] = useState(0);
+  const [grandTotal, setGrandtotal] = useState(0.0);
```

Comment:
I think these 3 states are also inter-dependent on each other. If you managed to write the reducer above, it's a good exercise to rewrite this part to `useReducer` too.